community/kernel-api/src/main/java/org/neo4j/impl/kernel/api/SuspendableCursor.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.impl.kernel.api;
+
+/**
+ * After invoking {@link #suspend()} this cursor is closed and needs to be re-initialized.
+ *
+ * A cursor can be (re-)initialized by passing a position to the {@linkplain #resume(CursorPosition) resume method}.
+ *
+ * @param <Position> the type of position used by this cursor.
+ */
+interface SuspendableCursor<Position extends CursorPosition<Position>> extends Cursor
```

Comment:
We introduced this while remembering the use case of someone wanting to scan nodes over multiple transactions.